packages/flutter/lib/src/material/tabs.dart

```diff
@@ -760,543 +352,478 @@ class TabBar<T> extends Scrollable implements AppBarBottomWidget {
-    _initSelection(newSelection);
-
-    assert(config.labels.isNotEmpty);
-    assert(Material.of(context) != null);
-
-    ThemeData themeData = Theme.of(context);
-    Color backgroundColor = Material.of(context).color;
-    Color indicatorColor = config.indicatorColor ?? themeData.indicatorColor;
-    if (indicatorColor == backgroundColor) {
-      // ThemeData tries to avoid this by having indicatorColor avoid being the
-      // primaryColor. However, it's possible that the tab bar is on a
-      // Material that isn't the primaryColor. In that case, if the indicator
-      // color ends up clashing, then this overrides it. When that happens,
-      // automatic transitions of the theme will likely look ugly as the
-      // indicator color suddenly snaps to white at one end, but it's not clear
-      // how to avoid that any further.
-      indicatorColor = Colors.white;
-    }
+      final int previousIndex = _controller.previousIndex;
 
-    final TextStyle textStyle = themeData.primaryTextTheme.body2;
-    final Color selectedLabelColor = config.labelColor ?? themeData.primaryTextTheme.body2.color;
-    final Color labelColor = selectedLabelColor.withAlpha(0xB2); // 70% alpha
-
-    List<Widget> tabs = <Widget>[];
-    bool textAndIcons = false;
-    int tabIndex = 0;
-    for (TabLabel label in config.labels.values) {
-      tabs.add(_toTab(label, tabIndex++, labelColor, selectedLabelColor));
-      if (label.hasText && label.hasIcon)
-        textAndIcons = true;
+      if (_controller.indexIsChanging) {
+        assert(_currentIndex != previousIndex);
+        wrappedTabs[_currentIndex] = new _TabStyle(
+          animation: _changeAnimation,
+          selected: true,
+          labelColor: config.labelColor,
+          child: wrappedTabs[_currentIndex],
+        );
+        wrappedTabs[previousIndex] = new _TabStyle(
+          animation: _changeAnimation,
+          selected: false,
+          labelColor: config.labelColor,
+          child: wrappedTabs[previousIndex],
+        );
+      } else {
+        wrappedTabs[_currentIndex] = new _TabStyle(
+          animation: kAlwaysCompleteAnimation,
+          selected: true,
+          labelColor: config.labelColor,
+          child: wrappedTabs[_currentIndex],
+        );
+      }
     }
 
-    Widget contents = new DefaultTextStyle(
-      style: textStyle,
-      child: new _TabBarWrapper(
-        children: tabs,
-        selectedIndex: _selection?.index,
-        indicatorColor: indicatorColor,
-        indicatorRect: _indicatorRect,
-        textAndIcons: textAndIcons,
-        isScrollable: config.isScrollable,
-        onLayoutChanged: _layoutChanged
-      )
+    // Add the tap handler to each tab. If the tab bar is scrollable
+    // then give all of the tabs equal flexibility so that their widths
+    // reflect the intrinsic width of their labels.
+    for(int index = 0; index < config.tabs.length; index++) {
+      final int tabIndex = index;
```

Comment:
You don't actually need to capture the index into a local in Dart, unlike most other algol languages.  Each iteration of the loop gets a fresh binding for index.